linked-list/linked-list.c

```diff
@@ -202,56 +202,26 @@ int main(int argc, char *argv[])
   #pragma omp target device(BIGPULP_SVM) map(to: vertices[0:n_vertices], n_vertices) \
     map(tofrom: n_successors_max)
   {
-    unsigned sync = 0;
+    unsigned n_vertices_local       = hero_tryread((unsigned int *)&n_vertices);
+    unsigned n_successors_max_local = hero_tryread((unsigned int *)&n_successors_max);
+    vertex * vertices_local         = (vertex *)hero_tryread((unsigned int *)&vertices);
 
-    #pragma omp parallel default(none) shared(vertices, n_vertices, n_successors_max, sync) \
-      num_threads(2)
+    #pragma omp parallel firstprivate(vertices_local, n_vertices_local) \
+      shared(n_successors_max_local) num_threads(1)
```

Comment:
Ciao @alessandrocapotondi, is this `omp parallel` with `num_threads(1)` needed? Or could we directly continue with
```c
unsigned n_successors_tmp = 0;
#pragma omp for
for (...) { ... }
```
Same question applies two more times in this PR.

Reply:
@accuminium this `num_threads(1)` is needed to enable correct execution as, at the moment, the linker fails when using `reduction` and `atomic` in the target. See comit `3f97c419f606c227dc26dce171f72820aa816fca`.